core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -190,25 +190,52 @@ oppia.directive('questionsList', [
               controller: [
                 '$scope', '$uibModalInstance',
                 function($scope, $uibModalInstance) {
-                  $scope.selectedSkillIds = [];
+                  var fillSkillDifficulties = function(length) {
+                    $scope.selectedSkillDifficulties = [];
+                    for (var i = 0; i < length; i++) {
+                      $scope.selectedSkillDifficulties.push('0.3');
+                    }
+                  };
+                  $scope.selectedSkillIds = linkedSkillIds;
+                  $scope.selectedSkillDescriptions = [];
+                  fillSkillDifficulties(ctrl.newQuestionSkillIds.length);
+                  $scope.selectDifficulty = selectDifficulty;
                   $scope.skillSummaries = allSkillSummaries;
-                  $scope.skillSummaries.forEach(function(summary) {
-                    summary.isSelected = false;
-                  });
+                  if ($scope.skillSummaries) {
+                    $scope.skillSummaries.forEach(function(summary) {
+                      summary.isSelected = false;
+                    });
+                  }
 
                   $scope.selectOrDeselectSkill = function(skillId, index) {
                     if (!$scope.skillSummaries[index].isSelected) {
                       $scope.selectedSkillIds.push(skillId);
+                      $scope.selectedSkillDescriptions.push(
+                        $scope.skillSummaries[index].getDescription());
                       $scope.skillSummaries[index].isSelected = true;
                     } else {
                       var idIndex = $scope.selectedSkillIds.indexOf(skillId);
                       $scope.selectedSkillIds.splice(idIndex, 1);
+                      $scope.selectedSkillDescriptions.splice(idIndex, 1);
```

Comment:
You know, I am getting a bit worried about this, because it's all too easy for some developer to later update one variable and not the other.

Does "id + description" (and possibility "+difficulty") make sense as a domain object and if so, should we define one? It would be better to have just a single list with the data tightly associated.

Reply:
I don't know how useful it would be as a domain object but I agree it should be tightly coupled. What do you think about adding them as separate objects here itself rather than creating a domain object?

Reply:
This one I haven't yet done because I think we should discuss this a bit

Reply:
Sorry, what do you mean by "separate objects"?

I think it would probably be better to make a domain object in order to couple the two things together properly. Having them be separate causes potential for "updating one but not the other", and we typically avoid passing dicts around because the set of properties is not guaranteed to be fixed (devs can inadvertently add extra properties midway).

Reply:
I have created a domain object for it.